rskj-core/src/test/java/co/rsk/peg/federation/FederationFactoryTest.java

```diff
@@ -0,0 +1,210 @@
+            version = federationPostRSKIP201PostRSKIP284PostRSKIP293.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+            assertNotEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PostRSKIP293);
+            assertNotEquals(federationPostRSKIP201PostRSKIP284PreRSKIP293, federationPostRSKIP201PostRSKIP284PostRSKIP293);
+        }
+
+        @Test
+        void differentNonStandardErpFederations_areNotEqualFeds_butHaveSameNonStandardFedFormat_mainnet() {
+            int version;
+
+            // in mainnet it should build non standard fed with csv unsigned big endian
+            when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);
+            ErpFederation federationPostRSKIP201PreRSKIP284 = createNonStandardErpFederation();
+            version = federationPostRSKIP201PreRSKIP284.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+
+            // should build non standard fed with csv unsigned big endian
+            when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+            ErpFederation federationPostRSKIP201PostRSKIP284PreRSKIP293 = createNonStandardErpFederation();
+            version = federationPostRSKIP201PostRSKIP284PreRSKIP293.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+            assertEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PreRSKIP293);
+
+            // build non standard fed
+            when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+            ErpFederation federationPostRSKIP201PostRSKIP284PostRSKIP293 = createNonStandardErpFederation();
+            version = federationPostRSKIP201PostRSKIP284PostRSKIP293.getFormatVersion();
+
+            assertEquals(NON_STANDARD_ERP_FEDERATION.getFormatVersion(), version);
+            assertNotEquals(federationPostRSKIP201PreRSKIP284, federationPostRSKIP201PostRSKIP284PostRSKIP293);
+            assertNotEquals(federationPostRSKIP201PostRSKIP284PreRSKIP293, federationPostRSKIP201PostRSKIP284PostRSKIP293);
+        }
+
+        @Test
+        void p2shErpFederation_haveP2shFedFormat() {
+            ErpFederation federation;
+            int version;
+
+            // mainnet
+            federation = createP2shErpFederation();
+            version = federation.getFormatVersion();
+            assertEquals(P2SH_ERP_FEDERATION.getFormatVersion(), version);
+
+            // testnet
+            bridgeConstants = BridgeTestNetConstants.getInstance();
+            networkParameters = bridgeConstants.getBtcParams();
+            federation = createP2shErpFederation();
+            version = federation.getFormatVersion();
+            assertEquals(P2SH_ERP_FEDERATION.getFormatVersion(), version);
+        }
+
+    }
+
+    public Federation createStandardMultisigFederation() {
```

Comment:
These 3 methods should probably be private, right?

Reply:
done